mopet/mopet.py

```diff
@@ -547,11 +547,10 @@ def _read_group_as_dict(self, group):
         for array in group:
             if isinstance(array, tables.array.Array):
                 value = array.read()
-                if not isinstance(value, np.ndarray):
-                    value = np.array(value)
-                # convert 0-dim arrays to floats
-                if value.ndim == 0:
-                    value = np.float(value)
+                # unwrap 0-dim arrays
+                if type(value) is np.ndarray and value.ndim == 0:
+                    value = array.dtype.type(value)
+
```

Comment:
Great! I did this because I wanted to make sure that every data type can be actually stored in the table, but your solution is much better! I suppose ints, floats etc, can be stored directly, without having to convert them to arrays, correct?

Reply:
Yes they can be stored directly! I covered the non-array cases in the test to make sure.